### ISSUE TITLE
Cross entropy loss criterion applied to probabilities rather than scores

### ISSUE THREAD

**Issue by kawu** [ShenMinX/point-gen#1]
From the [documentation](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss):
```
The input is expected to contain raw, unnormalized scores for each class.
```

In `decoder.py` (`baseline` branch):
```python
    p_vocab = torch.softmax(self.v(rnn_out.squeeze()), 1) # batch x 1 x hidden_size -> batch x vocab_size
    ...
    output = p_vocab # batch x vocab_size
    return output, rnn_hid, attn
```
In `main.py`:
```python
    output, rnn_hid, attn = model_decoder(enc_out, rnn_hid, dec_input, enc_input, attn)
    ...
    p_step_loss = criterion(output, target[:,i])
```
